hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2296,6 +2292,11 @@ public static HoodieMetadataColumnStats mergeColumnStatsRecords(HoodieMetadataCo
       return newColumnStats;
     }
 
+    // If new column stats is tight bound, then discard the previous column stats
+    if (newColumnStats.getIsTightBound() && !prevColumnStats.getIsTightBound()) {
```

Comment:
why does it matter if prev is tight bound or no? if newColStats is TightBound, we should just choose the new one and return right?